apps/mysql.go

```diff
@@ -34,6 +36,8 @@ type MetricsReceiverMySql struct {
 
 	Password string `yaml:"password" validate:"omitempty"`
 	Username string `yaml:"username" validate:"omitempty"`
+
+	ApplicationVersion string `yaml:"application_version" validate:"omitempty"`
```

Comment:
[Optional] Have we explored ways of doing this fallback automatically, without user input? E.g., modifying the upstream mysql receiver to detect whether new-style metrics are unavailable and issue the old-style queries?

Reply:
We have explored it, but when I discussed it with some opentelemetry community members there was some resistance to adding the archaic language that MySQL chose to replace (`slave status`) for a MySQL version that's nearly EOL, so we chose to avoid the uphill battle and implement in a way independent of the MySQL receiver.

Reply:
What if we just issued old-style SQL queries unconditionally from the Ops Agent-generated configs? Would that cause a lot of log spam on MySQL 8.x?

(I'm just trying to avoid adding new configuration options if we can help it…)

Reply:
Due to the way it's implemented, it wouldn't log spam but it would query both ways on MySQL 8. We wouldn't have duplicate metrics because the filtering happens post-query in the mysql receiver so it wouldn't be problematic.

This new configuration option was the decision the ops team made, so I'll defer to @jefferbrecht and team, do we want to just _always_ configure a secondary receiver to get replication metrics?

Reply:
I might've had an exaggerated sense of badness in my mind re: allowing both queries on version 8 -- if it turns out to not be a big deal then we can reconsider that design choice.

If I understand right, are you proposing that we just disable the replica metrics in the MySQL OTel receiver and use a second receiver (`sqlqueryreceiver`) with `slave status` instead, which supposedly works on both versions? That'd avoid duplication even before filtering, right?

Reply:
That's correct, that should work. The query would still be executed on version 8, but the metrics would never be created from that query.

Reply:
Sounds good -- We're fine on our end to give this a try, hopefully it simplifies things.

Reply:
Re-implemented in the described way (no more configuration, always collect replication metrics with `slave status` regardless of version), awaiting tests in kokoro.